code/game/machinery/_machinery.dm

```diff
@@ -301,6 +311,28 @@
 	set waitfor = FALSE
 	return PROCESS_KILL
 
+/**
+ * Generates heat on the turf that the machine is located on
+ * Args:
+ * - amount: The amount of heat to create
+ */
+/obj/machinery/proc/generate_heat(amount)
+	if(temperature_ignore_atmos) // skips all atmos related mechanics for machines
+		return TRUE
+
+	if(amount <= 0) //just in case
+		return FALSE
+
+	var/turf/local_turf = loc
+	if(!istype(local_turf)) // double check if this is even neccessar
+		return FALSE
+
+	var/datum/gas_mixture/enviroment = local_turf.return_air()
+	enviroment.temperature += amount / enviroment.heat_capacity()
```

Comment:
You should add some sort of checks to make sure it won't get to ridiculously high temperatures from excessively low heat capacity. Could make it thermal conduction based with an imaginary object at a given temperature and heat capacity, or you could have the servers itself contribute its own heat capacity to the calculation.